packages/compass-connections-navigation/src/navigation-item.tsx

```diff
@@ -251,6 +251,9 @@ export function NavigationItem({
           isActive={isActive}
           isFocused={isFocused}
           isExpanded={!!item.isExpanded}
+          hasDefaultAction={
+            item.type !== 'connection' || item.connectionStatus === 'connected'
```

Comment:
Ideally this would be co-located with or derived from the implementation in `onDefaultAction` in `connections-navigation-tree.tsx`, but that is not trivial because of the virtualization.